Account/FirefoxAccount.swift

```diff
@@ -264,6 +335,12 @@ open class FirefoxAccount {
         return advance().map { newState in
             if newState.label == FxAStateLabel.married {
                 if let married = newState as? MarriedState {
+                    
+                    if AppConstants.MOZ_FXA_PREFERENCE {
+                        // Upon an account being married, download the latest FxA account profile data.
```

Comment:
Technically it's "When the account enters the `Married` state".

But I'm not sure this is the right time to do this. This means we will be hitting the account server, and potentially downloading the profile image, every single time we launch the app — or perhaps every time we sync!

Why not do this before we load the Settings UI?

If we're going to do this routinely, we should be caching the image.